pycloudlib/azure/cloud.py

```diff
@@ -699,9 +723,13 @@ def launch(
         inbound_ports=None,
         username: Optional[str] = None,
         resource_group: Optional[util.AzureParams] = None,
-        network_security_group: Optional[util.AzureCreateParams] = None,
-        virtual_network: Optional[util.AzureCreateParams] = None,
-        subnet: Optional[util.AzureCreateParams] = None,
+        network_security_group_: Optional[util.AzureCreateParams] = None,
```

Comment:
Is there a style rationale for the trailing underscore?

Reply:
I ran into an issue with mypy, just added the _ to differentiate it with other variables in the function. It was just a hack around the issue.

Reply:
This is a fairly pedantic issue, but mangling variable names to make linters happy is sidestepping the issue. It would be more appropriate to rename the argument names to more closely align to what they actually are (not to embed the type of the resource within the name - but to have a more accurate definition). The arguments of `network_security_group_` and `virtual_network_` do not represent those SDK objects, but rather the configuration for them.

I would recommend renaming to either `nsg_params`/`vnet_params`/`subnet_params`. Likewise for the `ip_addresses` and `network_interface_clients` - these are all arguments which encapsulate configuration for the respective resource type - not the resource itself.